pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -485,6 +485,27 @@ func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance, opti
 			params.MigrateDisks = copyDisks
 			params.MigrateDisksSet = true
 		}
+
+		domainSpec, err := util.GetDomainSpecWithFlags(dom, 0)
+		if err != nil {
+			log.Log.Object(vmi).Reason(err).Error(fmt.Sprintf("Live migration failed."))
+			l.setMigrationResult(vmi, true, fmt.Sprintf("%v", err), "")
+			return
+		}
+
+		if registrar := cli.NewDomainEventDeviceRemovedRegistrar(l.virConn, dom); registrar != nil {
+			sriovDevices := sriov.FilterHostDevices(domainSpec)
+
+			err = sriov.WaitHostDevicesToDetach(registrar, sriovDevices, 30*time.Second, func() error {
+				return sriov.DetachHostDevices(dom, sriovDevices)
+			})
+			if err != nil {
+				log.Log.Object(vmi).Reason(err).Error(fmt.Sprintf("Live migration failed."))
+				l.setMigrationResult(vmi, true, fmt.Sprintf("%v", err), "")
+				return
+			}
+		}
```

Comment:
This needs to be placed in a dedicated function and processed only if the feature-gate is opened (#4874).